packages/dev/core/src/Rendering/IBLShadows/iblShadowsAccumulationPass.ts

```diff
@@ -207,17 +207,22 @@ export class _IblShadowsAccumulationPass {
         // Need to set all the textures first so that the effect gets created with the proper uniforms.
         this._setOutputTextureBindings();
 
-        let counter = 0;
-        this._scene.onBeforeRenderObservable.add(() => {
-            counter = 0;
-        });
-        this._scene.onAfterRenderTargetsRenderObservable.add(() => {
-            if (++counter == 2) {
-                if (this.enabled && this._outputTexture.isReady()) {
-                    this._setOutputTextureBindings();
-                    this._outputTexture.render();
+        this._renderPipeline.onVoxelizationCompleteObservable.addOnce(() => {
+            let counter = 0;
+            this._scene.onBeforeRenderObservable.add(() => {
+                counter = 0;
+            });
+            // onAfterRenderTargetsRenderObservable is called twice during a frame and we only want to render
+            // on the second call, after the scene has been rendered to the GBuffer.
+            this._scene.onAfterRenderTargetsRenderObservable.add(() => {
+                if (++counter == 2) {

```

Comment:
I still don't like this counter logic and I feel like it might be related to the problem. The `onAfterRenderTargetsRenderObservable` fires twice during a frame and I only want to run these passes once, after the geometry buffer is rendered and before the main render. Is there a better way to do this?

Reply:
I don't think this is related to the problem, although I do agree it's a bit ugly...

The second `onAfterRenderTargetsRenderObservable` may not be notified if there is no camera in the scene, but I don't think this is a problem(?)

At some point, we will need to port the IBL shadows to the frame graph, which will allow us to get rid of this workaround.

Reply:
About this is there no way to actually hook on the GBuffer onAfterRender observable ?

Reply:
Ah, I didn't think about it, I was looking at hooks at the scene level!

Yes, @MiiBond you could hook on `geometryBufferRenderer.getGBuffer().onAfterRenderObservable`

Reply:
Oh, this is excellent! Thank you.

Reply:
I've made this change now.